### PR TITLE
Document reserved partitioning names

### LINKED ISSUE #218: `GraphPartitioner` uses names that aren't in the table of reserved names in `design.rst`
cc @matthiasdiener

### PR DESCRIPTION
**Pull request by matthiasdiener**
Fixes #218

### CODE REVIEW

**Comment by matthiasdiener** [comment]
Side note: I couldn't find pt_shp or pt_data used in the code.

File: doc/design.rst [hunk 1]
```diff
@@ -136,6 +136,10 @@ Reserved Identifiers
         names of :class:`~pytato.array.DataWrapper` arguments that are
         not supplied by the user.
 
+    -   ``_pt_part_ph``: Used to automatically generate identifiers for
+        names of :class:`~pytato.array.Placeholder` that represent data
+        transport across parts of a partitioned DAG.
```
Comment (inducer) [inline doc/design.rst:141]:
```suggestion
        transport across parts of a partitioned DAG.
        (cf. :func:`find_partitions`)
```

**Review by inducer — APPROVED** [review]

**Comment by inducer** [comment]
> Side note: I couldn't find `pt_shp` or `pt_data` used in the code.

`_pt_data` should be used here:

https://github.com/inducer/pytato/blob/7455b3a62dfd95b4b4a972b1afeea69957679fc5/pytato/codegen.py#L162

`_pt_shp` is unused for now because we don't yet support computed shapes (well, er, at all). That's why this is unused.

**Comment by inducer** [comment]
Thx!

**Pull request merged**